twisted/python/_shellcomp.py

```diff
@@ -350,16 +351,17 @@ def writeHeader(self):
         This is the start of the code that calls _arguments
         @return: L{None}
         """
-        self.file.write('#compdef %s\n\n'
-                        '_arguments -s -A "-*" \\\n' % (self.cmdName,))
+        self.file.write(b'#compdef %s\n\n'
+                        b'_arguments -s -A "-*" \\\n'
+                        % self.cmdName.encode('utf-8'))
```

Comment:
The build works on Python 2.7 and Python 3.5, but it fails on Python 3.3 and Python 3.4 here:

https://travis-ci.org/twisted/twisted/jobs/144663141#L2296

That is because % formatting for bytes was not implemented until Python 3.5 : https://www.python.org/dev/peps/pep-0461/

To move this along and get it working on Python 3.3 and Python 3.4, I suggest that you
not use % formatting here.